eth/filters/filter.go

```diff
@@ -74,6 +74,9 @@ func (self *Filter) SetTopics(topics [][]common.Hash) {
 func (self *Filter) Find() vm.Logs {
 	latestHash := core.GetHeadBlockHash(self.db)
 	latestBlock := core.GetBlock(self.db, latestHash, core.GetBlockNumber(self.db, latestHash))
+	if latestBlock == nil {
+		return vm.Logs{}
+	}
```

Comment:
Do we still need this after the database write-order fixes? The panic is a nice way to catch the invariant and not let it slip by undetected.